libzecale/circuits/aggregator_circuit_wrapper.tcc

```diff
@@ -49,7 +62,8 @@ libsnark::protoboard<libff::Fr<wppT>> aggregator_circuit_wrapper<
     NumProofs>::get_constraint_system() const
 {
     libsnark::protoboard<libff::Fr<wppT>> pb;
-    aggregator_gadget<nppT, wppT, nsnarkT, wverifierT, NumProofs> g(pb);
+    aggregator_gadget<nppT, wppT, nsnarkT, wverifierT, NumProofs> g(
+        pb, num_inputs_per_nested_proof);
```

Comment:
Why not adding this gadget back to the attributes of the class? That would remove the need to instantiate it in all functions.

Reply:
It's a little bit more complex, since the protoboard etc is also local to each function.  I've done that refactor in a later PR, but here I've just removed the as-yet unused member.

Reply:
Hum... I am not following here. What was wrong with keeping `aggregator_g` as a private attribute of this class?

Reply:
The member `aggregator_g` was unused, I think.  `g` was being instantiated locally in each function in the original code.  To make it work, we need to keep `pb` as a member as well and move quite a bit of code around, otherwise AFAICT we'd probably get memory trampling and all kinds of issues, and I didn't want to do that refactor here (a separate memory bug in prover_server also gets shown up by that change, which needs to be fixed at the same time).

So for now there is no point in having only a shared pointer to the gadget as a member.  The switch to fully cache everything in members is done in a refactor of this code in a later PR.

Reply:
>  The switch to fully cache everything in members is done in a refactor of this code in a later PR.

Alright, let's discuss on this other PR then, we can keep things like that for now.